addons/mrp/models/mrp_bom.py

```diff
@@ -242,24 +242,27 @@ def create(self, vals_list):
             production._link_bom(res[0])
         return res
 
+    def _set_outdated_bom_in_productions(self):
+        # Searches for MOs using these BoMs to notify them that their BoM has been updated.
+        list_of_domain_by_bom = []
+        for bom in self:
+            domain_by_products = [('product_id', 'in', bom.product_tmpl_id.product_variant_ids.ids)]
+            if bom.product_id:
+                domain_by_products = [('product_id', '=', bom.product_id.id)]
+            domain_for_confirmed_mo = AND([[('state', '=', 'confirmed')], domain_by_products])
+            # Avoid confirmed MOs if the BoM's product was changed.
+            domain_by_states = OR([[('state', '=', 'draft')], domain_for_confirmed_mo])
+            list_of_domain_by_bom.append(AND([[('bom_id', '=', bom.id)], domain_by_states]))
+        domain = OR(list_of_domain_by_bom)
+        productions = self.env['mrp.production'].search(domain)
+        if productions:
+            productions.is_outdated_bom = True
```

Comment:
to be safe, do not default to searching on an empty domain (which will return all of the MOs)

```suggestion
        if list_of_domain_by_bom:
            domain = OR(list_of_domain_by_bom)
            productions = self.env['mrp.production'].search(domain)
            if productions:
                productions.is_outdated_bom = True
```